.github/workflows/ci.yml

```diff
@@ -164,15 +186,34 @@ jobs:
           fi
           echo "Upload successful!"
 
-      - name: Upload a Build Artifact
-        uses: actions/upload-artifact@v4
+  release:
+    name: Release - github
+    if: startsWith(github.ref, 'refs/tags/')
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download artifacts
+        uses: actions/download-artifact@v4
         with:
-          # Artifact name
-          name: ${{ matrix.os }}_${{ matrix.cc }}_${{ env.ZIP_FILENAME }}
-          # optional, default is artifact
-          # A file, directory or wildcard pattern that describes what to upload
-          path: ${{ github.workspace }}/extension/build/package/${{ matrix.os }}_${{ matrix.cc }}_${{ env.ZIP_FILENAME }}
-          # The desired behavior if no files are found using the provided path.
-          if-no-files-found: error
-          # Duration after which artifact will expire in days. 0 means using default retention.
-          retention-days: 14
+          path: ./gh-artifacts
+
+      - name: Package artifacts
+        working-directory: gh-artifacts
+        run: |
+          version=`echo $GITHUB_REF | sed "s/refs\/tags\///"`
+          ls -Rall
+          for platform in *; do
+            if [ -d "./$platform/" ]; then
+              cd "./$platform/"
+              tar -czf "../${{ github.event.repository.name }}-${platform,,}.tar.gz" -T <(ls -1)
+              cd -
+            fi
+          done
+
+      - name: Release packages
+        uses: svenstaro/upload-release-action@v2
```

Comment:
Not super comfortable with 3rd party action package, especially if they end up manipulating the repository secrets. Can't this be rewritten as a simple bash script ?